google/cloud/datastore/helpers.py

```diff
@@ -119,11 +124,12 @@ def entity_from_protobuf(pb):
     :rtype: :class:`google.cloud.datastore.entity.Entity`
     :returns: The entity derived from the protobuf.
     """
-
-    if not getattr(pb, "_pb", False):
-        # Coerce raw pb type into proto-plus pythonic type.
-        proto_pb = entity_pb2.Entity(pb)
-        pb = pb
+    if not isinstance(pb, entity_pb2.Entity):
+        # Coerce a dictionary into a raw protobuf
+        if not isinstance(pb, message.Message):
+            pb = entity_pb2.Entity.pb(**pb)
+        # Wrap the constructed protobuf with proto-plus dressing
+        proto_pb = entity_pb2.Entity.wrap(pb)
```

Comment:
We need to have test coverage for all three cases (dict, message, entity wrapper).

Reply:
I'm surprised that untested line didn't get flagged by CI. I'll get on this.

Reply:
Test added ✅